client/src/modules/grades/grades.js

```diff
@@ -0,0 +1,104 @@
+angular.module('bhima.controllers')
+.controller('GradeManagementController', GradeManagementController);
+
+GradeManagementController.$inject = [
+  'GradeService', 'ModalService',
+  'NotifyService', 'uiGridConstants', '$state', 'SessionService',
+];
+
+/**
+ * Grade Management Controller
+ *
+ * This controller is about the grade management module in the admin zone
+ * It's responsible for creating, editing and updating a grade
+ */
+function GradeManagementController(Grades, ModalService,
+  Notify, uiGridConstants, $state, Session) {
+  var vm = this;
+
+  // bind methods
+  vm.deleteGrade = deleteGrade;
+  vm.editGrade = editGrade;
+  vm.createGrade = createGrade;
+  vm.toggleFilter = toggleFilter;
+
+  // global variables
+  vm.gridApi = {};
+  vm.filterEnabled = false;
+
+  // options for the UI grid
+  vm.gridOptions = {
+    appScopeProvider  : vm,
+    enableColumnMenus : false,
+    fastWatch         : true,
+    flatEntityAccess  : true,
+    enableSorting     : true,
+    onRegisterApi     : onRegisterApiFn,
+    columnDefs : [
+      { field : 'code', displayName : 'FORM.LABELS.CODE', headerCellFilter : 'translate' },
+      { field : 'text', displayName : 'FORM.LABELS.DESIGNATION', headerCellFilter : 'translate' },
+      { field : 'basic_salary',
+        displayName : 'FORM.LABELS.BASIC_SALARY',
+        headerCellFilter : 'translate',
+        cellFilter : 'currency:'.concat(Session.enterprise.currency_id),
```

Comment:
I wonder if we should allow payroll to be in multiple currencies?  I can imagine that a hospital might want to pay some employees in USD and some in FC.

Reply:
In bhima 1.X the basic salary was always defined with the currency of the enterprise, but when the agents were paid the currency could be specified individually for each employee

Reply:
Okay, that sounds like a good default.  When we actually install somewhere, we'll wee what our clients wish.